### PR TITLE
Add UI tests for American Express

### PR DESCRIPTION
**Pull request by sestevens**
### Summary of changes

- Convert American Express integration tests into UI tests. 

We weren't able to figure out why, but our Amex integration tests were failing when run on fresh simulators, which meant they were failing pretty consistently in CI. We converted them into UI tests, and we're hoping this makes them more stable 🤞 

### Checklist

- ~Added a changelog entry~

- @sestevens 
- @scannillo

### CODE REVIEW

**Review by demerino — APPROVED** [review]
Weird that the integration tests fail

File: Demo/UI Tests/American Express UI Tests/AmericanExpress_UITests.swift [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+        app.launchArguments.append("-ClientToken")
+        app.launchArguments.append("-Integration:BraintreeDemoAmexViewController")
+        app.launch()
+
+        waitForElementToAppear(app.buttons["Valid card"])
+    }
+
+    func testValidCard_receivesRewardsBalance() {
+        app.buttons["Valid card"].tap()
+        waitForElementToAppear(app.buttons["45256433 Points, 316795.03 USD"], timeout: 10)
```
Comment (sarahkoop) [inline Demo/UI Tests/American Express UI Tests/AmericanExpress_UITests.swift:21]: Do these `waitForElementToAppear` statements need to be wrapped in some sort of assertion?  It's unclear if the tests will actually fail or just return false if the wait times out.

Comment (sestevens) [reply Demo/UI Tests/American Express UI Tests/AmericanExpress_UITests.swift:21]: The test will fail if the element isn't found in the given amount of time. [`waitForElementToAppear`](https://github.com/braintree/braintree_ios/blob/bf20534a349502c74603eeba8b61e42ba4801b47/Demo/UI%20Tests/Helpers/BTUITest.swift#L4) calls XCTest's `waitForExpectations` method, which fails the test if the expectation isn't fulfilled.

Comment (sarahkoop) [reply Demo/UI Tests/American Express UI Tests/AmericanExpress_UITests.swift:21]: Ah ok thanks for the explanation!

**Review by sarahkoop — APPROVED** [review]
🌟

**Pull request merged**